src/main/java/htmlpublisher/HtmlPublisher.java

```diff
@@ -76,10 +77,15 @@ private static void writeFile(ArrayList<String> lines, File path) throws IOExcep
 
     public ArrayList<String> readFile(String filePath) throws java.io.FileNotFoundException,
             java.io.IOException {
+        return readFile(filePath, this.getClass());
+    }
+    
+    public static ArrayList<String> readFile(String filePath, Class<?> publisherClass) 
+            throws java.io.FileNotFoundException, java.io.IOException {
         ArrayList<String> aList = new ArrayList<String>();
 
         try {
-            final InputStream is = this.getClass().getResourceAsStream(filePath);
+            final InputStream is = publisherClass.getResourceAsStream(filePath);
```

Comment:
> child classes may have other resources (e.g. Doxygen plugin could use HTML publisher with other header/footer)

If this is intending to provide an API to other plugins, it should *not* accept a `String` resource path and use reflection to find that resource (as the current code does). Should instead accept a `URL`.

At any rate, GitHub search turns up no current external uses in @jenkinsci.